src/PWManifest.cpp

```diff
@@ -7,44 +7,24 @@
 #include <PWManifest.h>
 #include <utils.h>
 
-PWManifest::ArgDetails::ArgDetails(const std::string& addr)
-{
-    size_t start = 0, end;
-
-    end = addr.find(':', start);
-
-    if (end == std::string::npos)
-    {
-        inheritFrom = addr;
-        return;
-    }
-
-    inheritFrom = addr.substr(0, end);
-
-    if (end + 1 < addr.size())
-    {
-        keyPaths.push_back(addr.substr(end + 1, addr.size()));
-    }
-}
-
 void PWManifest::reserve(std::size_t count)
 {
     argIDTable.reserve(count);
     argManifest.reserve(count);
 }
 
-void PWManifest::insert(std::string_view name, PWManifest::ArgDetails&& arg)
+PWManifest::ARG_ID PWManifest::insert(std::string_view name, PWManifest::ArgDetails&& arg)
 {
-    argManifest.emplace(counter, std::move(arg));
+    auto [it, result] = argManifest.emplace(counter, std::move(arg));
+    (void) result; // unused
     argIDTable.emplace(name, counter);
```

Comment:
Not clear if you actually intend this to be a noop if `name` was added before. Perhaps you meant an error in this case. I also don't understand why `argManifest` is a map of sequential ints as keys rather than using a vector for this.